pkgs/applications/misc/hello/default.nix

```diff
@@ -30,6 +32,9 @@ stdenv.mkDerivation (finalAttrs: {
 
   passthru.tests.run = callPackage ./test.nix { hello = finalAttrs.finalPackage; };
 
+  # We allow `hello.src` to be used in tests and examples.
+  passthru.src = finalAttrs.src;
```

Comment:
and I think we should move all those tests out of hello and not abuse it as a testing package.

Reply:
I think it's a good usage for hello

Reply:
hello is used in many guides to demonstrate how all of the nix tools work. It is great for this because it is small and still can do something. Also it is probably not installed on most systems. If we move all the tests out of it we know that those guides continue to work and can test whatever because it will not affect those.

Reply:
I've separated the fancy stuff into a new package `salve-mundi` based on `hello`. This way we can have our cake and eat it too. I've already removed some unnecessary bits from `hello`. I think it's at a nice level now, still avoiding `rec` and adding a test, but not overly complicated.